docs/reference/files/file-charmcraft-yaml.rst

```diff
@@ -0,0 +1,901 @@
+        # filesystem will be presented to the charm as normal.
+        # The block type requests a raw block device, typically disks or logical volumes.
+        type: filesystem | block
+
+        # (Optional) Description of the storage requested
+        description: <description>
+
+        # (Optional) The mount location for filesystem stores. For multi-stores
+        # the location acts as the parent directory for each mounted store.
+        location: <location>
+
+        # (Optional) Indicates if all units of the application share the storage
+        shared: true | false
+
+        # (Optional) Indicates if the storage should be made read-only (where possible)
+        read-only: true | false
+
+        # (Optional) The number of storage instances to be requested
+        multiple:
+            range: <n> | <n>-<m> | <n>- | <n>+
+
+        # (Optional) Minimum size of requested storage in forms G, GiB, GB. Size
+        # multipliers are M, G, T, P, E, Z or Y. With no multiplier supplied, M
+        # is implied.
+        minimum-size: <n>| <n><multiplier>
+
+        # (Optional) List of properties, only supported value is "transient"
+        properties:
+          - transient
+
+
+.. dropdown:: Example
+
+    .. literalinclude:: charmcraft-sample-charm.yaml
+        :start-at: storage: # Possible storage for the charm
+        :end-before: subordinate:
+
+.. _file-charmcraft-yaml-subordinate:
+``subordinate``
+===============
+
+**Status:** Optional.
+
+**Purpose:** Configures whether the charm is meant to be deployed as a subordinate to a principal charm.
+
+**Structure:**
+
+.. code::
+
+    subordinate: true | false
+
+**Example:**
+
+.. code::
+
+    subordinate: false
+
+.. _file-charmcraft-yaml-summary:
+``summary``
+===========
+
+**Status:** Required if the `type`_ key is set to ``charm``.
+
+**Structure:** A short, one-line description of the charm. No more than 78 characters.
+
+.. dropdown:: Example
+
+    .. code::
+
+        summary: A Juju charm to run a Traefik-powered ingress controller on Kubernetes.
+
+.. _file-charmcraft-yaml-terms:
+``terms``
+=========
+
+**Status:** Optional.
+
+**Purpose:** Lists the terms that any charm user agree to when they're using the charm.
+
+**Structure:** The list of terms:
+
+.. code::
+
+    terms:
+      - <term>
+
+.. dropdown:: Example
+
+    .. literalinclude:: charmcraft-sample-charm.yaml
+        :start-at: terms:
+        :end-before: title: My awesome charm
+
+.. _file-charmcraft-yaml-title:
+``title``
+=========
+
+**Status:** Optional, but recommended
+
+**Purpose:** A human-readable name for your charm
+
+.. dropdown:: Example
+
+    .. literalinclude:: charmcraft-sample-charm.yaml
+        :start-at: title:
+        :end-before: type:
+
+.. _file-charmcraft-yaml-type:
+``type``
+========
+
+**Status:** Required.
+
+**Purpose:** Indicates whether charmcraft will pack a charm or a bundle.
+
+**Structure:**
+
+*Type:* String.
+
+*Value:* ``charm`` or ``bundle``.
```

Comment:
```suggestion
**Type:** String.

**Value:** ``charm`` or ``bundle``.
```

Reply:
I had done it this way initially to match what was on the Discourse docs, but I'm happy to change it.